internal/layer/config.go

```diff
@@ -0,0 +1,85 @@
+package layer
+
+import (
+	"encoding/json"
+	"fmt"
+	cdl "github.com/mimiro-io/common-datalayer"
+	"strings"
+)
+
+const (
+	// dataset mapping config
+	TableName      = "table_name"
+	FlushThreshold = "flush_threshold"
+	AppendMode     = "append_mode"
+	SinceColumn    = "since_column"
+	EntityColumn   = "entity_column"
+)
+
+type PgsqlConf struct {
+	Hostname string `json:"host"`
+	Port     string `json:"port"`
+	Database string `json:"database"`
+	User     string `json:"user"`
+	Password string `json:"password"`
+	Schema   string `json:"schema"`
+}
+
+func newPgsqlConf(config *cdl.Config) (*PgsqlConf, cdl.LayerError) {
+	c := &PgsqlConf{}
+	nativeConfig := config.NativeSystemConfig
+	configJson, _ := json.Marshal(nativeConfig)
+	err := json.Unmarshal(configJson, c)
+
+	if err != nil {
+		return nil, cdl.Err(fmt.Errorf("could not unmarshal native system config because %s", err.Error()), cdl.LayerErrorInternal)
+	}
+
+	return c, nil
+}
+
+func (dl *PgsqlDatalayer) UpdateConfiguration(config *cdl.Config) cdl.LayerError {
+	// update database connection
+	dl.db, _ = newPgsqlDB(config)
```

Comment:
what happens to the old db.db? should it be closed properly before overwriting with a new instance?

Reply:
probably nothing good. fixed.